Lifecycle Workflows Overview (#18474)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/identitygovernance-lifecycleworkflows-overview.md

For this reason, Lifecycle Workflows can be referred to as the **Joiner-Mover-Leaver** (JML) workflow. Only the *joiner* and *leaver* processes are currently implemented.

The lifecycle workflows APIs in Microsoft Graph allow you to automate the Lifecycle Workflows capabilities for your organization. This article introduced the set of APIs that enable the Lifecycle Workflows service in Azure AD.

The lifecycle workflows APIs are defined in the OData subnamespace, microsoft.graph.identityGovernance.

[!INCLUDE [GDPR-related-guidance](../../includes/gdpr-msgraph-export-note.md)]

## Workflows

Workflows are containers for the processes involved in managing the life cycle of users in the organization. At their core are *tasks* and *execution conditions*.

+ [Tasks](#workflow-tasks) are specific actions that run automatically when a workflow is triggered.
+ [Execution conditions](#execution-conditions) define the *scope* of "who" and the *trigger* of "when" a workflow will run.

:::image type="content" source="../../../concepts/images/identitygovernance-workflow-parts.png" alt-text="Parts of a workflow API object.":::

> To create workflows, use one of the [predefined workflow templates](#workflow-templates), or integrate with Logic Apps to run custom workflows.

Reviewer: ```suggestion
To create workflows, we recommend using one of the [predefined workflow templates](#workflow-templates).
```

        "rule": "department eq 'R&D'"
    },
    "trigger": {
        "@odata.type": "#microsoft.graph.identityGovernance.timeBasedAttributeTrigger",
        "timeBasedAttribute": "employeeLeaveDateTime",
        "offsetInDays": -7
    }
}
```

When creating or updating a workflow, use the [workflowExecutionConditions resource type](identitygovernance-workflowexecutionconditions.md) to configure the execution conditions. Use this object to also configure a workflow that runs on-demand only.

### Create and manage workflows

After identifying the tasks and execution conditions that you want to define for your workflow, use the [workflow resource type](identitygovernance-workflow.md), and its associated methods to create and manage the workflow. You can create up to 50 workflows in a tenant. The category of the task must match the category of the workflow. Each workflow can have up to 25 tasks. Therefore:

+ A task supported for only the "leaver" workflow category can't be specified in a "joiner" workflow scenario, and vice versa.
+ A task supported for both "leaver" and "joiner" workflow categories can be specified in either a "joiner" or "leaver" workflow scenario.

> You can schedule a workflow to run based on the [tenant-wide schedule](#settings) or run it on-demand. For example, a daily workflow can provision access for new employees, while you can run a custom extension to immediately terminate an employee's access to sensitive resources after a security breach.

Reviewer: ```suggestion
You can schedule a workflow to run based on the [tenant-wide schedule](#settings) or run it on-demand. The tenant schedule can take care of regular new hires and terminations, while you can run a workflow on-demand to immediately terminate an employee's access in case of a sensitive event.
```
